Communitter/src/main/java/com/communitter/api/model/PostVote.java

```diff
@@ -0,0 +1,50 @@
+package com.communitter.api.model;
+
+import com.fasterxml.jackson.annotation.JsonBackReference;
+import jakarta.persistence.*;
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+import java.time.LocalDateTime;
+
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
+@Entity
+@Table(name="post_votes")
+public class PostVote {
+    @Id
+    @SequenceGenerator(
+            name="post_vote_sequence",
+            sequenceName = "post_vote_sequence",
+            allocationSize = 1
+    )
+    @GeneratedValue(strategy = GenerationType.SEQUENCE,
+            generator = "post_vote_sequence")
+    private Long id;
+
+    @Column(name = "created_at", nullable = false, columnDefinition = "TIMESTAMP(6)")
+    private LocalDateTime createdAt;
+
+    @Column(name = "is_upvote", nullable = false)
+    private boolean isUpvote;
+
+    @ManyToOne
+    @JoinColumn(name = "post_id", nullable = false)
+    @JsonBackReference("post-votes")
+    private Post post;
+
+    @ManyToOne
+    @JoinColumn(name = "user_id")
+    @JsonBackReference("user-votes")
+    private User user;
+
+    public PostVote(Long id, boolean isUpvote, Post postToVote, User user) {
```

Comment:
Allargs ve Noargs constructor annotationlarından sonra buna ihtiyaç var mı?

Reply:
I needed this constructor to set "createdAt" variable to "LocalDateTime.now()"  inside of the constructor. I also tried to use @PrePersist annotation in order to prevent writing a constructor but it failed in multiple calls by client side.

Reply:
@Builder.Default
@Column(name = "created_at", nullable = false)
private OffsetDateTime createdAt = OffsetDateTime.now();

if you need default value, you can use @Builder.Default annotation.

Reply:
Thanks for the hint! It is fixed